server/app/controllers/admin/AdminExportController.java

```diff
@@ -70,10 +77,30 @@ public Result exportProgram(Http.Request request) {
         formFactory
             .form(AdminProgramExportForm.class)
             .bindFromRequest(request, AdminProgramExportForm.FIELD_NAMES.toArray(new String[0]));
-    // TODO(#7087): Show an error if no program was selected.
-    // TODO(#7087): Return JSON representing the exported program.
-    return notFound(
-        String.format(
-            "Received ID: %s. Program export is not yet implemented", form.get().getProgramId()));
+
+    Long programId = form.get().getProgramId();
+    if (programId == null) {
+      // If they didn't select anything, just re-render the main export page.
+      return redirect(routes.AdminExportController.index().url());
+    }
+
+    // TODO(#7087): The export UI only shows active programs. Should we not download the program
+    // JSON here if the programId is actually for a draft program?
+    ProgramDefinition program;
+    try {
+      program = programService.getFullProgramDefinition(programId);
+    } catch (ProgramNotFoundException e) {
+      return badRequest(String.format("Program with ID %s could not be found", programId));
+    }
+
+    ErrorAnd<String, String> serializeResult = programMigrationService.serialize(program);
```

Comment:
**_very_  nit and not a blocker, but to consider in future PRs**: the pattern elsewhere is for this to be a noun, like `serializationResult`

Reply:
oh I just realized that's not a new class, just an `ErrorAnd`, nvm!